### PR TITLE
Sirius Command Center addition

### PR DESCRIPTION
**Pull request by twurman**
This pr adds v0.1 of the command center to the clarity ecosystem. The command center is capable of processing sirius requests involving the asr, imm, and qa services. It can receive requests from v0.1 of the sirius mobile application.

### CODE REVIEW

**Comment by jhauswald** [comment]
@llotneb @twurman @cdmeister:
- Correct command line arguments and handling is very important for this: there are many services to open and its easy to mess it up.
- Its good that its pretty standard (port to listen, port for cc) but better handling would be good.
- @llotneb: I can't compile the QADaemon.java:
```
QADaemon.java:102: error: local variable port is accessed from within inner class; needs to be declared final
          System.out.println("Starting qa at port " + port + "...");
                                                      ^
QADaemon.java:103: error: local variable server is accessed from within inner class; needs to be declared final
          server.serve();
          ^
2 errors
```
- I currently have no way to test besides that I can get imm and asr to register with the CC, would be good to have a ccclient somewhere here; that's going to disappear though right?

When going through all this, you can definitely feel that its been created by a disparate group of people. That's fine and probably inevitatble but we should have a standard method of README, building and running, I was jumping around either typing `make` or `make thrift` etc. Binaries should be called the same way: `imm-server`, `asr-server`, `asr-client`, etc.

It has a good flow to it, definitely a lot nicer than the shell + python scripts from back in the day :+1:  :100: ! I'm going to keep this open for now so we can keep talking about it and @llotneb can make his fix.

**Comment by llotneb** [comment]
Hi -- I cloned twurman/clarityeco on clarity04, checked out the clarityinc_dev branch, and ran compile-qa.sh from within sirius/, but my version compiled. That error message makes me think that there might be a versioning issue. What version of java do you have? I'm using 1.8.0.

**Comment by jhauswald** [comment]
Good catch, yea it could be a version issue: I'm using 1.7.0_79. I'll look into installing 1.8.0 and let you know

**Comment by llotneb** [comment]
What clarity server are you working on? I could fix my qa code to be backwards compatible if this is the issue. I don't think it will be a hard change to make.
Also, to be more specific, I'm using 1.8.0_45.

**Comment by jhauswald** [comment]
- Each service should have sample input committed (that works): commit the sample files for the command-center client (wav and image).
- ccclient should print the query sent (I'm not sure what the question was). I know its just a test client but this is going to be how we stress test the services. We should look into using [gtest](https://code.google.com/p/googletest/)
    - Maybe an option to just run all the tests would be good
- We should favor boost over get_opts (that switch statement doesn't look sustainable with more options) in the next release #27 
- Noticed it uses two different wav for test. Definitely commit whatever the test runs. and make the inputs a parameter, NOT HARDCODED. With boost you can set default inputs [see djinn](https://github.com/jhauswald/djinn/blob/master/tonic-suite/img/src/img-client.cpp#L44)


CommandCenter(client | server) documentation + readmes is going to be very important #23. After committing the test files and removing hardcoded inputs, I'll merge and tag all of this.

File: command-center/CmdCenterDaemon.cpp [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+
+	// initialize the image matching server
+	TThreadPoolServer server(processor, serverTransport, transportFactory, protocolFactory, threadManager);
+	cout << "Starting the command center server on port " << port << "..." << endl;
+	server.serve();
+	cout << "Done." << endl;
+	return 0;
+}
+
+std::string parseImgFile(const std::string& immRetVal)
```
Comment (jhauswald) [inline command-center/CmdCenterDaemon.cpp]: this is cool: could definitely be expanded to replace this or make a new fancy query.

**Pull request merged**